### DISCUSSION TITLE
Functionality, how to use

### DISCUSSION THREAD

**Discussion by hakha4** [timmbogner/Farm-Data-Relay-System#198] [Q&A] [answered]
I have a network with some sensors and a couple of relay controllers and everything works ok most of the time. Trying to get a better understanding how things work under the hood and I have a few questions that someone maybe have the answers to:

What is the use of `ping` ? (I know what ping is in general!) . How to catch responses ?

What is the purpuse and how to  use mqtt command  `fdrs/status` ?

There is a `ota.h` file but how to implement ? Only on GW ? or can it be used in controller/sensors ? (would be nice for units hard to reach physically)

Best Regards Håkan

**Comment by timmbogner** [comment]
Hi Håkan,
#
Pinging works as follows... It may possibly change soon, because the functionality is being worked on in the timekeeping pull request that's in development:
``` cpp
  int p = 0;
  p = pingFDRS(300);
  DBG(p);
```
First we initialize `p`, which will hold our ping time. Next we run `pingFDRS(300)` and set `p` to the result. You can then use that value for whatever you may need it for. The timeout (300ms) is how long the node will wait for a reply before moving on.

#
`fdrs/status` is currently only used once, to display a "hello, world!" message. In the future I plan to use this topic to post `DBG()` messages from the gateway over MQTT. I will get to it eventually!
#
OTA should work on gateways that use WiFi or ethernet.  I have an idea in mind that could maybe allow any FDRS device to be put into WiFi mode for OTA, but that's another thing on the to-do list!

#

**Reply by hakha4** [reply, marked as answer]
Thank's !
